java/core/src/java/org/apache/orc/impl/ReaderImpl.java

```diff
@@ -74,7 +74,7 @@
   protected List<OrcProto.StripeStatistics> stripeStatistics;
   private final int metadataSize;
   protected final List<OrcProto.Type> types;
-  private TypeDescription schema;
+  private final TypeDescription schema;
```

Comment:
Let's revert this change.

Reply:
Any reason for revert or you prefer this also goes into a separate PR?

Reply:
Yes, a separate PR, please.